github.go

```diff
@@ -156,13 +159,15 @@ func ParseUrlIntoGitHubRepo(url string, token string, instance GitHubInstance) (
 }
 
 // Download the release asset with the given id and return its body
-func DownloadReleaseAsset(repo GitHubRepo, assetId int, destPath string) *FetchError {
+func DownloadReleaseAsset(repo GitHubRepo, assetId int, destPath string, withProgress bool) *FetchError {
 	url := createGitHubRepoUrlForPath(repo, fmt.Sprintf("releases/assets/%d", assetId))
 	resp, err := callGitHubApi(repo, url, map[string]string{"Accept": "application/octet-stream"})
 	if err != nil {
 		return err
 	}
-
+	if !withProgress {
+		return nil
+	}
```

Comment:
Wait... Doesn't this return before writing the response to disk?

Reply:
yeah, this should just be moved to the beginning of the function. :D

Reply:
moved